pkg/config/setup/otlp.go

```diff
@@ -12,86 +12,66 @@ import (
 // OTLP configuration paths.
 const (
 	OTLPSection               = "otlp_config"
-	OTLPTracesSubSectionKey   = "traces"
-	OTLPTracePort             = OTLPSection + "." + OTLPTracesSubSectionKey + ".internal_port"
-	OTLPTracesEnabled         = OTLPSection + "." + OTLPTracesSubSectionKey + ".enabled"
-	OTLPLogsSubSectionKey     = "logs"
-	OTLPLogsEnabled           = OTLPSection + "." + OTLPLogsSubSectionKey + ".enabled"
+	OTLPTracePort             = OTLPSection + ".traces.internal_port"
+	OTLPTracesEnabled         = OTLPSection + ".traces.enabled"
+	OTLPLogsEnabled           = OTLPSection + ".logs.enabled"
 	OTLPReceiverSubSectionKey = "receiver"
 	OTLPReceiverSection       = OTLPSection + "." + OTLPReceiverSubSectionKey
-	OTLPMetricsSubSectionKey  = "metrics"
-	OTLPMetrics               = OTLPSection + "." + OTLPMetricsSubSectionKey
-	OTLPMetricsEnabled        = OTLPSection + "." + OTLPMetricsSubSectionKey + ".enabled"
-	OTLPTagCardinalityKey     = OTLPMetrics + ".tag_cardinality"
-	OTLPDebugKey              = "debug"
-	OTLPDebug                 = OTLPSection + "." + OTLPDebugKey
+	OTLPMetrics               = OTLPSection + ".metrics"
+	OTLPMetricsEnabled        = OTLPMetrics + ".enabled"
+	OTLPDebug                 = OTLPSection + "." + "debug"
 )
 
 // OTLP related configuration.
 func OTLP(config pkgconfigmodel.Setup) {
-	config.BindEnvAndSetDefault(OTLPTracePort, 5003)
-	config.BindEnvAndSetDefault(OTLPMetricsEnabled, true)
-	config.BindEnvAndSetDefault(OTLPTracesEnabled, true)
-	config.BindEnvAndSetDefault(OTLPLogsEnabled, false)
+	config.BindEnv("otlp_config.grpc_port") // TODO OTLP team: add default value
+	config.BindEnv("otlp_config.http_port") // TODO OTLP team: add default value
 
 	// NOTE: This only partially works.
 	// The environment variable is also manually checked in comp/otelcol/otlp/config.go
-	config.BindEnvAndSetDefault(OTLPTagCardinalityKey, "low", "DD_OTLP_TAG_CARDINALITY")
+	config.BindEnvAndSetDefault("otlp_config.metrics.tag_cardinality", "low", "DD_OTLP_TAG_CARDINALITY")
```

Comment:
Is there a reason for removing the `OTLPTagCardinalityKey` constant (and the others in this file)?

Reply:
It makes it very hard to grep and find definition for a setting. using the full key is also what every other teams in the agent uses.